backend/tracim_backend/models/data.py

```diff
@@ -1139,19 +1142,26 @@ def owner(self, value: User) -> None:
     def owner(cls) -> InstrumentedAttribute:
         return ContentRevisionRO.owner
 
-    @hybrid_property
+    @property
     def children(self) -> ['Content']:
         """
         :return: list of children Content
         :rtype Content
         """
-        # Return a list of unique revisions parent content
-        # TODO - G.M - 2018-10-24 - Set is just here to avoid distinct object,
-        # check if possible to do this better, probably using database sorting
-        # and sql distinct.
-        revisions = set([revision.node for revision in self.children_revisions])
-        revisions = sorted(revisions, key=lambda r: r.revision_id)
-        return revisions
+        # TODO - G.M - 2019-02-08 - Refactor this code to be more efficient
+        all_childrens_revisions = object_session(self).query(ContentRevisionRO).filter( ContentRevisionRO.parent_id == self.content_id).all()
+        all_up_to_date_revisions_result = object_session(self).query(
+            func.max(ContentRevisionRO.revision_id)
+            ).group_by(ContentRevisionRO.content_id).all()
+
+        all_up_to_date_revisions_ids = [value[0] for value in all_up_to_date_revisions_result]
+
+        valid_children = []
+        for revision in all_childrens_revisions:
+            if revision.revision_id in all_up_to_date_revisions_ids:
+               valid_children.append(revision)
```

Comment:
note : in the original code you returned revision.node (which is a Content), here you seems to return ContentRevisionRO

Reply:
You're right. I have to fix this.

Reply:
done;